File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/ThreadUtils.java
package me.ycdev.android.lib.common.utils;

import android.os.Looper;

import java.util.Set;

@SuppressWarnings({"unused", "WeakerAccess"})
public class ThreadUtils {
    public static boolean isMainThread() {
        return Looper.myLooper() == Looper.getMainLooper();
    }

    public static boolean isThreadRunning(long tid) {
        Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
        for (Thread t : threadSet) {
            if (t.getId() == tid) {
                return true;
            }
        }
        return false;
    }
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/ThreadUtils.kt
package me.ycdev.android.lib.common.utils

import android.os.Looper

object ThreadUtils {
    val isMainThread: Boolean
        get() = Looper.myLooper() == Looper.getMainLooper()

    fun isThreadRunning(tid: Long): Boolean {
        val threadSet = Thread.getAllStackTraces().keys
        for (t in threadSet) {
            if (t.id == tid) {
                return true
            }
        }
        return false
    }

    fun sleep(millis: Long) {
        try {
            Thread.sleep(millis)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }
}
